pkg/proxy/reverseproxy_test.go

```diff
@@ -0,0 +1,116 @@
+package proxy_test
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/hellofresh/janus/pkg/proxy"
+	"github.com/hellofresh/janus/pkg/router"
+	"github.com/hellofresh/janus/pkg/test"
+	stats "github.com/hellofresh/stats-go"
+	"github.com/stretchr/testify/assert"
+)
+
+var tests = []struct {
+	description         string
+	method              string
+	url                 string
+	expectedContentType string
+	expectedCode        int
+}{
+	{
+		description:         "Get example route",
+		method:              "GET",
+		url:                 "/example",
+		expectedContentType: "application/json; charset=utf-8",
+		expectedCode:        http.StatusOK,
+	}, {
+		description:         "Get invalid route",
+		method:              "GET",
+		url:                 "/invalid-route",
+		expectedContentType: "text/plain; charset=utf-8",
+		expectedCode:        http.StatusNotFound,
+	},
+	{
+		description:         "Get one posts - strip path",
+		method:              "GET",
+		url:                 "/posts/1",
+		expectedContentType: "application/json; charset=utf-8",
+		expectedCode:        http.StatusOK,
+	},
+	{
+		description:         "Get one posts - append path",
+		method:              "GET",
+		url:                 "/append",
+		expectedContentType: "application/json; charset=utf-8",
+		expectedCode:        http.StatusOK,
+	},
+}
+
+func TestSuccessfulProxy(t *testing.T) {
+	ts := test.NewServer(createRegisterAndRouter())
+	defer ts.Close()
+
+	for _, tc := range tests {
+		res, err := ts.Do(tc.method, tc.url)
+		assert.NoError(t, err)
+		if res != nil {
+			defer res.Body.Close()
+		}
+
+		assert.Equal(t, tc.expectedContentType, res.Header.Get("Content-Type"))
+		assert.Equal(t, tc.expectedCode, res.StatusCode, tc.description)
+	}
+}
+
+func createProxyDefinitions() []*proxy.Definition {
+	return []*proxy.Definition{
+		&proxy.Definition{
+			ListenPath:  "/example/*",
+			UpstreamURL: "http://www.mocky.io/v2/58c6c60710000040151b7cad",
```

Comment:
It's definitely bad idea to make real http requests in unit tests. I understand that this is really useful thing, but not part of unit tests. Let's think about introducing integration tests here and run them separately.